sip/execution_framework_interface/docker_api/client/docker_client.py

```diff
@@ -0,0 +1,470 @@
+                               ' all the services.')
+
+        service_list = self._client.services.list()
+        for s_list in service_list:
+            services.append(s_list.id)
+        return services
+
+    def get_service_details(self, service_id):
+        """ Get details of a service
+
+        Args:
+            service_id (string): List of service ID
+
+        Returns:
+            dict, details of the service
+        """
+
+        # Raise an exception if we are not a manager
+        if not self._manager:
+            raise RuntimeError('Only the Swarm manager node can retrieve all'
+                               ' the services details.')
+
+        service = self._client.services.get(service_id)
+        return service.attrs
+
+    def get_volume_list(self):
+        """ Get a list of docker volumes
+
+        Only the manager nodes can retrieve all the volumes
+
+        Returns:
+            list, all the names of the volumes in swarm
+        """
+        volumes = []
+
+        # Raise an exception if we are not a manager
+        if not self._manager:
+            raise RuntimeError('Only the Swarm manager node can retrieve'
+                               ' all the services.')
+
+        volume_list = self._client.volumes.list()
+        for v_list in volume_list:
+            volumes.append(v_list.name)
+        return volumes
+
+    def get_volume_details(self, volume_name):
+        """ Get details of the volume
+
+        Args:
+            volume_name (string): Name of the volume
+
+        Returns:
+            dict, details of the volume
+        """
+
+        if volume_name not in self.get_volume_list():
+            raise RuntimeError('No such volume found: ', volume_name)
+
+        volume = self._client.volumes.get(volume_name)
+        return volume.attrs
+
+    ###########################################################################
+    # Update functions
+    ###########################################################################
+
+    def update_labels(self, node_name, labels):
+        """Update label of a node
+
+          Args:
+            node_name (string): Name of the node.
+            labels (dict): Label to add to the node
+        """
+
+        # Raise an exception if we are not a manager
+        if not self._manager:
+            raise RuntimeError('Only the Swarm manager node can update '
+                               'node details.')
+
+        # Node specification
+        node_spec = {'Availability': 'active',
+                     'Name': node_name,
+                     'Role': 'manager',
+                     'Labels': labels}
+        node = self._client.nodes.get(node_name)
+        node.update(node_spec)
+
+    ###########################################################################
+    # Parsing functions
+    ###########################################################################
+
+    def parse_services(self, service_config, service_name):
```

Comment:
- is this a private function? if so prefix with an underscore
- It looks like this function does a bit more than parse services (the docstring also suggests this), maybe rename to `_parse_compose_file`?